### DISCUSSION TITLE
JLCPCB version of project (for fully-assembled boards)

### DISCUSSION THREAD

**Discussion by kpfleming** [hrvach/deskhop#52] [Show and tell]
I've made a fork of this project at https://github.com/kpfleming/deskhop; the changes I made were to make it easy to order fully-assembled boards from JLCPCB.

Changes:
* Changed to TI isolator.
* Changed to Jing USB-A connector (different size from Molex connector so PCB modified to accommodate it); this required importing a new KiCAD symbol/footprint/shape.
* Redrew USB D+/D- signal pairs using KiCAD 7 'differential pair' router to keep them balanced.
* Added BOM and CPL files for upload to JLCPCB.
* Renamed source files so that output files would not be the same names as the originals.

**Comment by hrvach** [comment]
Great job man, well done!

TI is better-priced and has better specs, so it's also the first choice for me as well in the revision board. I'm waiting for the boards to arrive so I can verify everything works as intended before updating the repo, it's been a month now and still nothing :-|

Differential pairs were done like this to compensate a bit for the difference in length of the traces on the pico pcb itself, so the total would match and more or less hit the target impedance. Since it's not hi speed (480) but full speed (12) at best, and given how short the traces are, I believe this shouldn't be a problem either way...

Are these USB-A connectors available in Europe anywhere? Let me know how your assembled boards turn out from JLC as well as the final price of an assembled board - any scenario where the final board is cheaper and more accessible to people is great! Thank you!

**Reply by kpfleming** [reply]
Those connectors can be purchased from [LCSC](lcsc.com), but that's probably the only source for them.

Once the boards arrive I'll post pictures and a review here; I ordered ten and almost all are committed to various friends/colleagues who wanted one so they will go fast :)

Also, I have contemplated placing the RP2040 and flash directly on the board and skipping the 'Pico' carrier board; that would be a much larger project (and definitely not for hand assembly). The RP2040 is around $1 in volume, the flash chip is cheap, and the rest of the parts are too... so that change would probably reduce the assembled board cost by $4-5.

**Reply by hrvach** [reply]
I'm thinking of doing a standalone board, but just as you said it would be impossible to hand-solder. I figured out how to bootstrap the second rp2040 through SWD from the first, so having just 1 flash and 1 quartz by outputting the clock from the first one should cut the cost even further, but it's still less of a difference than I expected. I'll still probably have a go at it after I improve some of the software stuff first...

**Comment by kpfleming** [comment]
I've just put the first board from this batch (assembled at JLCPCB) into production on my desk and it's working perfectly! I've got 9 more boards to program and test this weekend which will be going out to friends once their cases arrive.

**Reply by ehssame** [reply]
Hi, any chance you have any spare board and case left, happy to pay for the board/case and the shipping ? Alternatively, could you guide me please on where I need to order (JLCPCB.com I am guessing ?) and what file(s) to use for the order (newbie here so pardon the ignorance). many thanks to you and hrvach for doing this. Thx

**Reply by kpfleming** [reply]
Yes, I do have one board + case available, in the eastern US. Feel free to contact me via email at "kevin at km6g dot us", or Matrix at "at kevin colon km6g dot us", or Signal at "kpfleming dot 06".

**Comment by kpfleming** [comment]
So because I'm a little crazy I decided to attempt to design a version where all the parts are directly on the board (no Picos involved). I have a schematic done but not yet verified, it's attached. I've begun working on the PCB but that will probably take a couple of weeks to get right.

It's intended to be 100% software compatible with the existing design, but has lots of hardware changes:

- All SMT components (not really suitable for hand assembly, the passive components are 0402 packages)
- USB Type C connectors for input and output
- A second USB connector on the host side (both A and B) which can be used solely for powering the board (data lines are not connected); if power is supplied to that connector then power from the other connector (connected to the computer) is not used. This is intended for situations like mine where the board is connected to two laptops and they aren't always on the desk or powered up together, but the Deskhop should still be functional.
- Simpler power circuit (no buck-boost converter as it is not needed)

The projected BOM cost for this configuration is about US$6.25, which means fully assembled boards in quantity 10 orders would probably be about US$10 each. Because of the connector changes the case would also need some redesign, but it should be straightforward.

Since I'm enjoying doing this I'm going to continue with the PCB layout and then will publish it all in a branch in my fork; I have no idea whether I'll ever get any boards like this manufactured, but if it turns out there is interest in a low-cost version that is fully assembled it might be worth going this direction.
[DeskHop_JLCPCB.pdf](https://github.com/hrvach/deskhop/files/14550527/DeskHop_JLCPCB.pdf)

**Reply by hrvach** [reply]
Great work, would you consider combining the efforts? I have been working on a similar thing with just one flash and quartz, bootstrapping over SWD, etc...

**Reply by kpfleming** [reply]
Maybe... I'd like to get this one to the place where it's operational first, I might even get a small number of boards made to prove the design is correct. The cost for the additional parts is so low I'm not sure the added complexity of sharing the flash is worthwhile, and I really do like the isolation feature of the existing board design. In any case I am very much happy to collaborate :-)

**Reply by hrvach** [reply]
Isolation would stay, just get moved to one usb output. But I also like the idea of one additional USB, that's a great one!

**Reply by kpfleming** [reply]
More progress on this flavor: I've tentatively named it DeskHop-C (because it uses Type-C connectors).

The schematic is basically done, and I'm pretty confident it is correct.

The PCB layout is also basically done, and I'm pretty confident in most of it except for the RP2040 itself; that will require some review by people who have more knowledge of layout for that part in particular. The rest of the parts on the board were fairly easy to handle.

In order to avoid unnecessarily complex trace routes I had to change some of the GPIO assignments (for the UART connections and the device USB port); if this design goes forward I'll add some mechanism in the software build process to indicate the board 'flavor' so the proper GPIO assignments can be compiled in.

I would say this is at best 'alpha' quality right now, but I've pushed everything to https://github.com/kpfleming/deskhop/tree/type-c-jlcpcb. The schematic PDF is updated but the Gerber files are not, so if you want to review the PCB you'll need to grab the KiCAD files and open them there. I still need to spend some time updating the schematic with all of the manufacturing details, so that the BOM and CPL can be generated properly.

**Reply by kpfleming** [reply]
I've continued working on the 'DeskHop-C', fixing lots of little issues and improving the layout. I've put the design up for review on ["r/PrintedCircuitBoard"](https://www.reddit.com/r/PrintedCircuitBoard/comments/1bh7edd/review_request_my_first_smt_pcb_rp2040_based/) to see if people with experience with these parts can see any issues.

**Reply by kpfleming** [reply]
I've finally gotten a 3-D rendering as well; enjoy :-)

![image](https://github.com/hrvach/deskhop/assets/305938/062417eb-b4c9-4654-ab2d-a07d18ae2657)

**Reply by hrvach** [reply]
This looks very nice, congrats! I've finally managed to receive my v1.1 board redesign, so while not this nice, should be published soon.

**Reply by kpfleming** [reply]
What sort of isolation mechanism are you thinking about using in the single-flash-chip design? Connecting the GPIOs from one RP2040 to the SWD on the other one will break the isolation between them, and I can't find any inexpensive/practical ways to provide isolation for SWD itself.

This morning I considered creating a design where the isolation happened directly at each output USB port, allowing the rest of the board to be a single zone (with its own power input), but same issue there... the only USB isolators I can find are so expensive that they would make the board cost dramatically higher.

**Comment by p3lim** [comment]
Just used your fork to send off an order at JLCPCB, worked pretty smoothly apart from the Picos being rotated 270° incorrectly (easy enough to fix in their placing confirmation tool).

As for the USB-C version, I kind of disagree with this being the default (or atleast provide both A and C schematics), as most keyboards and mice still use USB-A as it's way more secure and they are components that usually never/rarely get disconnected in common setups.

The merging of the Pico and the PCB is a great idea, especially for getting finished products pre-assembled like from JLCPCB, but again should be optional if your fork would be merged in.

**Reply by kpfleming** [reply]
> Just used your fork to send off an order at JLCPCB, worked pretty smoothly apart from the Picos being rotated 270° incorrectly (easy enough to fix in their placing confirmation tool).

I'm glad it was useful for you! Indeed, the placement (CPL) file is incorrect; when I produced that I wasn't aware of the "JLCPCB Fabrication Toolkit" plugin for KiCAD, which I'll be using now to produce correct fabrication files. When I had my boards produced all of the components except the USB-A connectors were incorrectly placed but easy to fix. I'll get the CPL file in this branch updated soon.

A word of caution: I ordered 10 boards, 9 of them worked perfectly, 1 has a non-operational mouse port. This could just be a bad solder joint as the Picos had to be hand-soldered, and I'll do a closer inspection of that board to see if I can get it working.

> As for the USB-C version, I kind of disagree with this being the default (or atleast provide both A and C schematics), as most keyboards and mice still use USB-A as it's way more secure and they are components that usually never/rarely get disconnected in common setups.

I have no desire for the USB-C version to be the 'default', just an option. In my case both of the computers have USB-C ports (and I'd prefer to use them) as do my keyboard and mouse (when I use the mouse in wired mode). I'm happy to maintain this 'flavor' as a fork, or as a separate branch in the repo if the maintainer wishes that.

**Reply by p3lim** [reply]
Received my 5 boards from JLCPCB and they worked just fine.

**Comment by hrvach** [comment]
> What sort of isolation mechanism are you thinking about using in the single-flash-chip design? Connecting the GPIOs from one RP2040 to the SWD on the other one will break the isolation between them, and I can't find any inexpensive/practical ways to provide isolation for SWD itself.
> 
> This morning I considered creating a design where the isolation happened directly at each output USB port, allowing the rest of the board to be a single zone (with its own power input), but same issue there... the only USB isolators I can find are so expensive that they would make the board cost dramatically higher.

My idea was to isolate one USB port only, with either ADuM3160 or a significantly cheaper WCH CH315G. This would have a similar isolation effect but without the different power domains for the two picos. This would resolve the issue where you need to have both devices plugged in, but also split the 500 mA power budget between the rp2040s and keyboard + mouse, which I wouldn't consider a problem 10 years ago, but nowadays with all the RGB stuff going around...

Also, ADuM is quite expensive and WCH is only available through LCSC and similar suppliers, not really guaranteeing for a stable supply... so it's still on the drawing board :)

**Reply by kpfleming** [reply]
OK, we've had the same idea then :-) I'll investigate the CH315G out of curiosity, but I'm going to continue providing a 'power-only' connector so that the higher power budget can be supported.

**Reply by kpfleming** [reply]
Hmm... those parts are still on the expensive side (for a board whose other components cost under $6 in total!). In spite of that, the overall value of this product is so high that using isolators seems reasonable, I may experiment with a design that has double isolation (output ports isolated from center, with a center power supply and no power from the output ports).
